src/lib/FreeC/Backend/Agda/Syntax.hs

```diff
@@ -3,7 +3,71 @@
 
 module FreeC.Backend.Agda.Syntax
   ( module Agda.Syntax.Concrete
+  , module Agda.Syntax.Common
+  , module Agda.Syntax.Position
+  , name
+  , intLiteral
+  , simpleImport
+  , lambda
+  , qname
+  , qname'
   )
 where
 
 import           Agda.Syntax.Concrete
+import           Agda.Syntax.Position
+import           Agda.Syntax.Common
+
+import qualified Agda.Syntax.Common            as Agda
+import qualified Agda.Syntax.Concrete          as Agda
+import qualified Agda.Syntax.Literal           as Agda
+import qualified Agda.Syntax.Position          as Agda
+
+-------------------------------------------------------------------------------
+-- Identifiers                                                               --
+-------------------------------------------------------------------------------
+
+-- | Creates a (not qualified) Agda variable name from a 'String'.
+name :: String -> Agda.Name
+name ident = Agda.Name Agda.NoRange Agda.InScope [Agda.Id ident]
+
+-- | Create a qualified name given a local variable 'Name' and a list of module
+--   'Name's.
+qname :: Agda.Name -> [Agda.Name] -> QName
+qname n = foldr Agda.Qual (Agda.QName n)
```

Comment:
Since the module name is usually written before the name, I'd swap the arguments of `qname`.